src/core/load-fragment.c

```diff
@@ -3830,7 +3830,13 @@ static int load_from_path(Unit *u, const char *path) {
                         if (r >= 0)
                                 break;
                         filename = mfree(filename);
-                        if (r != -ENOENT)
+
+                        /* ENOENT means that the file is missing or is a dangling symlink.
+                         * ENOTDIR means that one of paths we expect to be is a directory
+                         * is not a directory, we should just ignore that.
+                         * EACCES means that the directory or file permissions are wrong.
+                         */
+                        if (!IN_SET(r, -ENOENT, -ENOTDIR, -EACCES))
```

Comment:
Let's say that the admin does e.g. 'chmod o= /usr/local/lib/systemd' or an equivalent operation, for example because of a botched backup restore. Most likely this would be unintentional, but currently this means a hard failure to load any units, and probably an unbootable system. We should treat this the same as if the directory was completely missing.